src/main/java/com/yy5/employee/service/EmployeeService.java

```diff
@@ -31,5 +30,29 @@ public Employee insert (String name, int age ) {
         employeeMapper.insert(employee);
         return employee;
     }
+
+    public void update(int employeeNumber, String name, Integer age) throws EmployeeNotFoundException {
+        Employee employee = findEmployee(employeeNumber);
+        if (employee == null) {
+            throw new EmployeeNotFoundException("employeeNumberが" + employeeNumber + "の社員は存在しません");
+        }
```

Comment:
ServiceのfindEmployee使う場合、employeeNumberが見つからない場合にServiceのfindEmployeeの中でExceptionがthrowされますね！そうすると、このupdateメソッドのnullチェックは絶対に通らないことになります。
ソースコードの見通しが悪くなるので、以下のように直接MapperのfindByEmployeeを呼んであげる方が良いですね！
```suggestion
        Employee employee =  this.employeeMapper.findByEmployee(employeeNumber)
                .orElseThrow(() -> new EmployeeNotFoundException("EmployeeNumber:" + employeeNumber +" is not found"));
```

Reply:
承知しました。
可読性を高めるため、updateメソッドを修正しました。
[390159c](https://github.com/YYprogramer/employee/pull/14/commits/390159c5b6ae7f547bec939d223d31787e1cc252)